src/combinatorics/simple_graph/connectivity.lean

```diff
@@ -383,8 +383,11 @@ def reachable_setoid : setoid V := setoid.mk _ G.reachable_is_equivalence
 the `simple_graph.reachable` relation. -/
 def connected_component := quot G.reachable
 
-/-- A graph is connected is every pair of vertices is reachable from one another. -/
-def connected : Prop := ∀ (u v : V), G.reachable u v
+/-- A graph is preconnected if every pair of vertices is reachable from one another. -/
+def preconnected : Prop := ∀ (u v : V), G.reachable u v
+
+/-- A graph is connected if it's preconnected and contains at least one vertex. -/
+def connected : Prop := G.preconnected ∧ nonempty V
```

Comment:
Let's make this conjunction non-anonymous:
```suggestion
@[protect_proj]
structure connected : Prop :=
(preconnected : G.preconnected)
(nonempty : nonempty V)
```
We can still write `h.1` for `h.preconnected` if we're feeling lazy.  (The `protect_proj` gives `connected.preconnected` the `protected` attribute, which keeps it from conflicting with `simple_graph.preconnected` inside the `simple_graph.connected` namespace, if we ever write code inside it.)